xilem_core/src/views/map_action.rs

```diff
@@ -0,0 +1,118 @@
+use core::marker::PhantomData;
+
+use crate::{DynMessage, Mut, View, ViewId, ViewPathTracker};
+
+/// A view that maps a child [`View<State,ChildAction,_>`] to [`View<State,ParentAction,_>`] while providing mutable access to `State` in the map function.
+///
+/// This is very similar to the Elm architecture, where the parent view can update state based on the action message from the child view.
+pub struct MapAction<
```

Comment:
Previously named `AdaptState` I'm still not sure about the naming, but I *think* `MapState` makes more sense?